build/foss/build.sh

```diff
@@ -1,21 +1,104 @@
 #!/usr/bin/env bash
-set -euox
+set -euxo pipefail
 
-source "${WORKDIR}/ci/archs.sh"
+source "${WORKDIR}/lib-versions.env"
 
 declare -A targets=(
   [amd64]=x86_64-unknown-linux-gnu
-  [aarch64]=aarch64-unknown-linux-gnu
   [i386]=i686-unknown-linux-gnu
+  [arm64]=aarch64-unknown-linux-gnu
   [armhf]=armv7-unknown-linux-gnueabihf
   [armel]=arm-unknown-linux-gnueabi
 )
 
-pushd "${WORKDIR}/build/foss"
-for arch in "${ARCHS[@]}"; do
-  target="${targets[$arch]}"
-  cargo build --target "${target}" --release
-  mkdir -p "${WORKDIR}/bin/deps/foss/${arch}"
-  ln -frs "${WORKDIR}/build/foss/target/${target}/release" "${WORKDIR}/bin/deps/foss/${arch}/latest"
-done
-popd
+function clone_if_absent() {
+  if [[ $# -ne 3 ]]; then
+    echo "Three parameters are required for clone_if_absent function:"
+    echo "clone_if_absent <repo_url> <tag/version> <destination_dir>"
+    echo -e "\t repo_url - URL to the repository being cloned"
+    echo -e "\t tag/version - repository tag\version you want to clone"
+    echo -e "\t destination_dir - where to clone the repo to"
+    echo "You passed ${#} arg(s): ${*}"
+    exit 1
+  fi
+  repo_url=$1
+  version=$2
+  dst_arch_dir=$3
+  repo_name="${repo_url##*/}"
+  repo_name="${repo_name%.git}"
+
+  if [[ ! -d "${dst_arch_dir}/${repo_name}" ]]; then
+    pushd "${dst_arch_dir}"
+    git clone --branch "${version}" "${repo_url}"
```

Comment:
Please check if this is doable with branches, tags and commit checksums

Reply:
This didn't work, but I updated it to allow checkout to any revision